validation/genesis/genesis-predeploy_test.go

```diff
@@ -145,100 +151,107 @@ func testGenesisPredeploys(t *testing.T, chain *ChainConfig) {
 			continue
 		}
 
-		// Load the genesis for the chain we are validating from the superchain module
-		g, err := LoadGenesis(chainId)
-		require.NoError(t, err)
-
 		if _, ok := g.Alloc[MustHexToAddress(address)]; !ok {
 			t.Fatalf("expected an account at %s, but did not find one", address)
 		}
 
-		{ // code validation
-			account := g.Alloc[MustHexToAddress(address)]
-			gotByteCode, err := LoadContractBytecode(account.CodeHash)
-			require.NoError(t, err)
-
-			var wantByteCodeHex string
-			artifactName, ok := predeployArtifactNames[address]
-			var cd *ContractData
-			if ok {
-				// it is a predeploy, we need to perform masking
-				// in order to validate it
-
-				data, err := os.ReadFile(path.Join(contractsDir, "forge-artifacts", artifactName+".sol", artifactName+".json"))
-				require.NoError(t, err)
+		validateCode(t, address, g, syntheticOPMainnetGenesis)
+		validateBalance(t, address, account, g)
+	}
+}
 
-				cd = new(ContractData)
-				err = json.Unmarshal(data, cd)
-				require.NoError(t, err)
-				wantByteCodeHex = cd.DeployedBytecode.Object
+func getDirOfThisFile() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("No caller information")
+	}
+	return filepath.Dir(filename)
+}
 
-				require.NoError(t, err)
+func validateCode(t *testing.T, address string, g *Genesis, syntheticOPMainnetGenesis *GenesisLite) {
+	account := g.Alloc[MustHexToAddress(address)]
+	if account.CodeHash == *new(Hash) {
+		// no validation needed if codeHash is set to zero value
+		return
+	}
+	gotByteCode, err := LoadContractBytecode(account.CodeHash)
+	require.NoError(t, err)
 
-				err = maskBytecode(gotByteCode, cd.DeployedBytecode.ImmutableReferences)
-				if err != nil {
-					t.Errorf("err masking bytecode for %s, %s", address, err)
-				}
+	var wantByteCodeHex string
+	artifactName, ok := predeployArtifactNames[address]
+	var cd *ContractData
+	if ok {
+		// it is a predeploy, we need to perform masking in order to validate it
+		filePath := path.Join(contractsDir, "forge-artifacts", artifactName+".sol", artifactName+".json")
+		data, err := os.ReadFile(filePath)
+		// Check if the error is due to the file not existing
+		if err != nil {
+			if os.IsNotExist(err) {
+				t.Logf("File not found: %s", filePath)
+				return
```

Comment:
I think this is ok because the set of hardcoded `predeployArtifactNames` probably doesn't apply to all chains, especially ones created long ago. But would be good to verify this. I was getting errors about `EAS.json` and `SchemaRegistry.json` files not existing. 

Also in the `monorepo-outputs.sh` script we are always generating outputs based on mainnet. Should we have an arg so it uses the op sepolia deploy-config if we are validating a sepolia chain?

Reply:
It might be that these contracts got moved into a legacy folder or something: if they are present in the synthetic genesis file, I would expect the source files + deployment artifacts to exit somewhere...

It might be that the address is just getting 1 wei allocated or something, and no actual code. So we can definitely have an escape hatch if there is empty bytecode in the synthetic genesis.

We should add a comment here for now and we can circle back and check.

Reply:
I put an escape hatch [here](https://github.com/ethereum-optimism/superchain-registry/pull/484/files#diff-0951f9e907e0bc03da7d9405099e27b8b3b006040f37ef12112c1cd0a7ff6b42R173-R176) for the case where the code hash is zero. Not sure if that fully covers it though

Reply:
Also, should [this line](https://github.com/ethereum-optimism/superchain-registry/pull/484/files#diff-0951f9e907e0bc03da7d9405099e27b8b3b006040f37ef12112c1cd0a7ff6b42R130) always use a synthetic version of `mainnet`? Or should it use a synthetic version of `sepolia` if we are validating a `sepolia` chain?

Reply:
> Also, should [this line](https://github.com/ethereum-optimism/superchain-registry/pull/484/files#diff-0951f9e907e0bc03da7d9405099e27b8b3b006040f37ef12112c1cd0a7ff6b42R130) always use a synthetic version of `mainnet`? Or should it use a synthetic version of `sepolia` if we are validating a `sepolia` chain?

We could try that, I'm not sure if it will make much difference?